File: exercises/advent2021/src/day2/part1.clj
(ns day2.part1
  (:require
   [clojure.string :as s]))

(def input-file
  (str "data/day2.txt"))

(defn parse-line
  [line]
  (let [[move amount] (s/split line #" ")]
    [move (Integer/parseInt amount)]))

(def input
  (->> input-file
       (slurp)
       (s/split-lines)
       (map parse-line)))

(defn step
  [[position depth]
   [move amount]]
  (case move
    "forward" [(+ position amount) depth]
    "down" [position (+ depth amount)]
    "up" [position (- depth amount)]))

(def final-destination
  (reduce step [0 0] input))

(def result
  (apply * final-destination))

(println result)

File: exercises/advent2021/src/day7/part1.py
def parse(filename: str) -> list[int]:
    with open(filename) as file:
        line = file.read()
        return list(map(int, line.split(',')))

def solve(input: list[int]) -> int:
    input.sort()
    end = len(input) - 1

    cost = 0
    for i in range(end//2):
        cost += input[end-i] - input[i]

    return cost

def main():
    input = parse('data/day7.txt')
    result = solve(input)
    print(result)

if __name__ == '__main__':
    main()